std/conv.d

```diff
@@ -1013,7 +1013,22 @@ if (!(is(S : T) &&
     !isEnumStrToStr!(S, T) && !isNullToStr!(S, T)) &&
     !isInfinite!S && isExactSomeString!T)
 {
-    static if (isExactSomeString!S && value[0].sizeof == ElementEncodingType!T.sizeof)
+    static if (is(S == struct) &&
+               __traits(hasMember, S, "toString") &&
+               isSomeFunction!(value.toString) &&
+               !__traits(isDisabled, S.toString) &&
+               is(typeof(value.toString()) == string))
```

Comment:
```suggestion
               is(typeof(value.toString()) Ret : T) &&
               isSomeString!Ret
```
The condition here should be based on this one in `std.internal.format.write.hasToString`: https://github.com/dlang/phobos/blob/67afde4f10d41d4168663f839e053298b306564d/std/format/internal/write.d#L1896